### ISSUE TITLE
Support to add additional methods to exclude

### ISSUE THREAD

**Issue by webbower** [kodefox/class-autobind#3]
Some packages provide mixins (which I know are frowned upon but still unavoidable in certain cases) or subclasses of `React.Component` that create additional component lifecycle methods. It would be good to allow the addition of other method names to be excluded in the binding to avoid possible weird behavior. Some example libraries I know of (because we use them):
- React Router (v1 provided a mixin that added a lifecycle method (`routerWillLeave`). Subsequent versions have changed the pattern).
- React Meteor Data provides a mixin that adds support for a lifecycle method (`getMeteorData`) to fetch data from a Meteor backend.

Maybe something like:

`autobind.excludeAdditional(['method1', 'method2', ...]) // globally add to REACT_EXCLUDE_METHODS`

or

`autobind(this, otherProto, { exclude: ['method1', 'method2', ...] }) // for one-off instances`

**Comment by sstur** [comment]
I totally agree. I'm open to either approach, alhough I like the second one better since it's less global. PR would be welcome.

**Comment by webbower** [comment]
@sstur Could be good to support both, although I think it might be better to support the global one if you're going to choose just one. Our site uses React + Meteor so the React Meteor Data mixin method appears a lot and the per-usage pattern would result in a lot of boilerplate. I would assume many people's projects would be like this, having a handful of mixin methods that appear throughout the codebase.

**Comment by sstur** [comment]
I'm down with that.

**Comment by cakeinpanic** [comment]
I suggest to add 'private method prefix' property
If i want to have private methods and start all of them with underscore for example

**Comment by zmorris** [comment]
I'd prefer to use the more-general [class-autobind](https://github.com/kodefox/class-autobind) instead of the more-specific [react-autobind](https://github.com/cassiozen/React-autobind), but I don't think I can until a way to exclude or blacklist functions is provided. Here's how they did it, although I think that `include` and `exclude` might be better keys than `bindOnly` and `wontBind`:

https://github.com/cassiozen/React-autobind#usage
